module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+#include <algorithm>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& imgToCompress) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedImg;
+ 
+    std::for_each(imgToCompress.begin(), imgToCompress.end(), [&compressedImg](const auto& row) {
+        uint8_t pixCount = 0;
+        for (auto it = row.begin(); it != row.end(); ++it) {
```

Comment:
Huge lambda :)

Reply:
Actually, that's the question I have to ask on discord or Q&A session - are there any considerations about how big the lambda should be? I am not sure if anything was said about it during the lesson?

Reply:
Ofc I will mention only most important things about lambda. There are a lot of information which wasn't told because you need to learn new things to understand this :)

And what about size. Lambda should be as small as possible.